src/App.tsx

```diff
@@ -1,24 +1,192 @@
+
+      return;
+    }
+
+    const data = {
+      id: 0,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    setTempTodo(data);
+    createTodoOnServer(data);
+  }, []);
+
+  const deleteTodo = useCallback((id: number) => {
+    setIdOfDeletedTodo(id);
+    deleteTodoFromServer(id);
+  }, []);
+
+  const deleteAllCompleted = useCallback(() => {
+    const completedTodoIds = todos
+      .filter(({ completed }) => completed)
+      .map(({ id }) => id);
+
+    setCompletedTodosID(completedTodoIds);
+
+    completedTodoIds.forEach(id => {
+      deleteTodoFromServer(id);
+    });
+  }, [todos]);
+
+  const isCompletedTodos = useCallback(() => {
+    return todos.some(({ completed }) => completed);
+  }, [todos]);
+
+  const closeErrorMessage = useCallback(() => {
+    setHasError(false);
+  }, []);
+
+  const handleSelect = useCallback((status: Filter) => {
+    setFilterOfTodo(status);
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterOfTodo) {
+      case Filter.Active:
+        return todos.filter(({ completed }) => !completed);
+      case Filter.Completed:
+        return todos.filter(({ completed }) => completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterOfTodo]);
+
+  useEffect(() => {
+    loadTodoFromServer();
+  }, []);
+
+  useEffect(() => {
+    let timeoutID: ReturnType<typeof setTimeout>;
+
+    if (hasError) {
+      timeoutID = setTimeout(() => {
+        setHasError(false);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(timeoutID);
+    };
+  }, [hasError]);
+
+  const countOfActiveTodos = useMemo(() => visibleTodos
+    .filter(({ completed }) => !completed)
+    .length, [visibleTodos]);
 
-export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoForm addTodo={addTodo} tempTodo={tempTodo} />
+
+        {!!visibleTodos.length && (
+          <TodoList
+            todos={visibleTodos}
+            tempTodo={tempTodo}
+            deleteTodo={deleteTodo}
+            idOfDeletedTodo={idOfDeletedTodo}
+            completedTodosID={completedTodosID}
+          />
+        ) }
+
+        {!!todos.length && (
+          <BottomBar
+            itemsCount={countOfActiveTodos}
+            selectedFilter={filterOfTodo}
+            onChange={handleSelect}
+            deleteAllCompleted={deleteAllCompleted}
+            showClearAllButton={isCompletedTodos()}
```

Comment:
You need just disable this button
```suggestion
            itemsCount={countOfActiveTodos}
            selectedFilter={filterOfTodo}
            onChange={handleSelect}
            onDeleteCompleted={deleteAllCompleted}
            isDisable={isCompletedTodos}
```